fastscape/tests/test_processes_tectonics.py

```diff
@@ -0,0 +1,163 @@
+    np.testing.assert_equal(p.bedrock_upward, sum_bedrock_forcing_vars)
+
+    sum_surface_forcing_vars = uplift + isostasy + surf_advect
+    np.testing.assert_equal(p.surface_upward, sum_surface_forcing_vars)
+
+    domain_rate = np.sum(sum_surface_forcing_vars) * area / dt
+    assert p._domain_rate() == domain_rate
+
+
+def test_surface_after_tectonics():
+    grid_shape = (3, 2)
+    topo_elevation = np.random.uniform(size=grid_shape)
+    forced_motion = np.random.uniform(size=grid_shape)
+
+    p = SurfaceAfterTectonics(topo_elevation=topo_elevation,
+                              forced_motion=forced_motion)
+
+    p.run_step()
+
+    expected = topo_elevation + forced_motion
+    np.testing.assert_equal(p.elevation, expected)
+
+
+@pytest.mark.parametrize('b_status, expected_mask, expected_uplift', [
+    (np.array(['fixed_value', 'fixed_value', 'fixed_value', 'fixed_value']),
+    np.array([[0., 0., 0., 0.],
+              [0., 1., 1., 0.],
+              [0., 0., 0., 0.]]),
+    np.array([[0., 0., 0., 0.],
+              [0., 50., 50., 0.],
+              [0., 0., 0., 0.]])),
+    (np.array(['looped', 'looped', 'looped', 'looped']),
+    np.ones((3, 4)),
+    np.full((3, 4), 50)),
+    (np.array(['core', 'core', 'core', 'core']),
+    np.ones((3, 4)),
+    np.full((3, 4), 50)),
+    (np.array(['fixed_value', 'looped', 'core', 'fixed_value']),
+    np.array([[0., 1., 1., 1.],
+              [0., 1., 1., 1.],
+              [0., 0., 0., 0.]]),
+    np.array([[0., 50., 50., 50.],
+              [0., 50., 50., 50.],
+              [0., 0., 0., 0.]]))
+])
+def test_block_uplift(b_status, expected_mask, expected_uplift):
+    rate = 5
+    grid = (3,4)
+    dt = 10.
+    f = FastscapelibContext(shape=grid, length=(10., 30.), ibc=1010)
+    f.initialize()
+    f.run_step(dt)
+
+    p = BlockUplift(rate=rate, 
+                    shape=grid,
+                    status=b_status,
+                    fs_context=f)
+
+    p.initialize()
+    np.testing.assert_equal(p._mask, expected_mask)
+
+    p.run_step(dt)
+    np.testing.assert_equal(p.uplift, expected_uplift)
+
+
+@pytest.mark.parametrize('x, expected_x_idx, expected_uplift', [
+    (2,
+     0,
+    np.full((3, 4), 30.)),
+    (np.array([1, 2, 3]),
+     1,
+     np.array([[20., 30., 30., 30.],
+               [20., 30., 30., 30.],
+               [20., 30., 30., 30.]]))
+])
+def test_two_blocks_uplift(x, expected_x_idx, expected_uplift):
+    x_pos = 1
+    rate_l = 2
+    rate_r = 3
+    grid = (3, 4)
+    dt = 10.
+
+    p = TwoBlocksUplift(x_position=x_pos,
+                        rate_left=rate_l,
+                        rate_right=rate_r,
+                        shape=grid,
+                        x=x)
+
+    p.initialize()
+    assert p._x_idx == expected_x_idx
+
+    p.run_step(dt)
+    np.testing.assert_equal(p.uplift, expected_uplift)
+
+
+@pytest.mark.parametrize('u, v, expected_surface_veffect, expected_bedrock_veffect', [
```

Comment:
We should really find a simple set of input values, e.g., set velocity values, `dt`, elevation and grid geometry such that we can test resuts (elevation values after advection) against some exact values (up to a precision tolerance, due to the non-exact algorithm and computing precision number).